TODO1/how-javascript-works-under-the-hood-of-css-and-js-animations-how-to-optimize-their-performance.md

```diff
@@ -172,174 +168,173 @@ animation.addEventListener('finish', function() {
 
-#### Bézier curves
+#### 贝塞尔曲线
 
-Let’s see hоw Bézier curves work. A Bézier curve takes four values, or more precisely it takes two pairs of numbers. Each pair describes the X and Y coordinates of a cubic Bézier curve’s control point. The starting point of the Bézier curve has a coordinate (0, 0) and the end coordinate is (1, 1). You can set both of the pair numbers. The X values for the two control points must be in the [0, 1] range, and each control point’s Y value can exceed the [0, 1] limit, although the spec isn’t clear by how much.
+让我们看一下贝塞尔曲线的工作原理。贝塞尔曲线有四个值，或者更确切地说，它需要两对数字。每对描述三次贝塞尔曲线控制点的 X 和 Y 坐标。贝塞尔曲线的起点坐标是 (0, 0)，终点坐标是 (1, 1)。两者都可以被设置。两个控制点的 X 值必须在 [0, 1] 范围内，并且每个控制点的 Y 值可以超过 [0, 1] 限制，尽管规范没有明确说超过多少。
 
-Even slight changes in the X and Y value of each control point gives you a completely different curve. Let’s take a look at two graphs of Bézier curves with points having close but different coordinates.
+即使每个控制点的 X 和 Y 值发生轻微变化，都会给你一个完全不同的曲线。我们来看看两个B贝塞尔曲线图，点的坐标相近但不同。
 
 ![](https://cdn-images-1.medium.com/max/800/1*2v7G1ZJ1C-y_mWHOYQfQKQ.png)
 
-and
+和
 
 ![](https://cdn-images-1.medium.com/max/800/1*P5nzyldL4rg36dZmt2RViQ.png)
 
-As you see, the graphs are quite different. The first control points have a
-(0.045, 0.183) vector difference, while the second control points have a (-0.427, -0.054) difference.
+如你所见，两个图区别比较大。两条曲线的第一个控制点的矢量差为 (0.045, 0.183)，第二个控制点差 (-0.427, -0.054)。
 
-This is how the CSS for the second curve looks like:
+第二条曲线的 CSS 写法如下：
 
-```
+```css
 transition: transform 500ms cubic-bezier(0.465, 0.183, 0.153, 0.946);
 ```
 
-The first two numbers are the X and Y coordinates of the first control point, and the second two numbers are the X and Y coordinates of the second control point.
+前两个数字是第一个控制点的 X 和 Y 坐标，后两个数字是第二个控制点的 X 和 Y 坐标。
 
-### Performance optimizations
+### 性能优化
 
-You should maintain 60fps whenever you are animating, otherwise it will negatively impact your users’ experience.
+无论何时动画，你都应该保持 60 fps，否则会对用户的体验产生负面影响。
 
-As everything else in the world, animations do not come free. Animating some properties is cheaper than others. For example, animating `width` and `height` of an element changes its geometry and may cause other elements on the page to move or change size. This process is called layout. We have discussed layout and rendering in more detail in one of our [previous posts](https://blog.sessionstack.com/how-javascript-works-the-rendering-engine-and-tips-to-optimize-its-performance-7b95553baeda).
+与世界上其他所有的东西一样，动画并不是免费的。动画一些属性比其他属性更便宜。例如，动画修改一个元素的 `width` 和 `height` 会改变它的形状，而且可能引起页面上其它元素的移动和形状改变。这个过程称为布局。我们在[之前的一篇文章](https://blog.sessionstack.com/how-javascript-works-the-rendering-engine-and-tips-to-optimize-its-performance-7b95553baeda)中已经详细讨论过布局和渲染。
```

Comment:
https://github.com/xitu/gold-miner/blob/master/TODO1/how-javascript-works-the-rendering-engine-and-tips-to-optimize-its-performance.md